src/libs/OptionsListUtils.ts

```diff
@@ -1625,131 +1669,142 @@ function getValidReports(reports: OptionList['reports'], config: GetValidReports
+        isReportArchived: !!option.private_isArchived,
+    });
 
-        const shouldBeInOptionList = shouldReportBeInOptionList({
-            report,
-            chatReport,
-            currentReportId: topmostReportId,
-            betas,
-            doesReportHaveViolations,
-            isInFocusMode: false,
-            excludeEmptyChats: false,
-            includeSelfDM,
-            login: option.login,
-            includeDomainEmail,
-            isReportArchived: !!option.private_isArchived,
-        });
+    if (!shouldBeInOptionList) {
+        return false;
+    }
 
-        if (!shouldBeInOptionList) {
-            continue;
-        }
+    const isThread = option.isThread;
+    const isTaskReport = option.isTaskReport;
+    const isPolicyExpenseChat = option.isPolicyExpenseChat;
+    const isMoneyRequestReport = option.isMoneyRequestReport;
+    const isSelfDM = option.isSelfDM;
+    const isChatRoom = option.isChatRoom;
+    const accountIDs = getParticipantsAccountIDsForDisplay(report);
 
-        const isThread = option.isThread;
-        const isTaskReport = option.isTaskReport;
-        const isPolicyExpenseChat = option.isPolicyExpenseChat;
-        const isMoneyRequestReport = option.isMoneyRequestReport;
-        const isSelfDM = option.isSelfDM;
-        const isChatRoom = option.isChatRoom;
-        const accountIDs = getParticipantsAccountIDsForDisplay(report);
+    if (excludeNonAdminWorkspaces && !isPolicyAdmin(option.policyID, policies)) {
+        return false;
+    }
 
-        if (excludeNonAdminWorkspaces && !isPolicyAdmin(option.policyID, policies)) {
-            continue;
-        }
+    if (isPolicyExpenseChat && report?.isOwnPolicyExpenseChat && !includeOwnedWorkspaceChats) {
+        return false;
+    }
+    // When passing includeP2P false we are trying to hide features from users that are not ready for P2P and limited to expense chats only.
+    if (!includeP2P && !isPolicyExpenseChat) {
+        return false;
+    }
 
-        if (isPolicyExpenseChat && report.isOwnPolicyExpenseChat && !includeOwnedWorkspaceChats) {
-            continue;
-        }
+    if (isSelfDM && !includeSelfDM) {
+        return false;
+    }
 
-        // When passing includeP2P false we are trying to hide features from users that are not ready for P2P and limited to expense chats only.
-        if (!includeP2P && !isPolicyExpenseChat) {
-            continue;
-        }
+    if (isThread && !includeThreads) {
+        return false;
+    }
 
-        if (isSelfDM && !includeSelfDM) {
-            continue;
-        }
+    if (isTaskReport && !includeTasks) {
+        return false;
+    }
 
-        if (isThread && !includeThreads) {
-            continue;
-        }
+    if (isMoneyRequestReport && !includeMoneyRequests) {
+        return false;
+    }
 
-        if (isTaskReport && !includeTasks) {
-            continue;
-        }
+    if (!canUserPerformWriteAction(report) && !includeReadOnly) {
```

Comment:
NAB put the pure boolean before the function call so JS short-circuits and doesn't run the function call unnecessarily:

```suggestion
    if (!includeReadOnly && !canUserPerformWriteAction(report)) {
```